src/main/java/com/minecolonies/coremod/blocks/BlockPaperwall.java

```diff
@@ -0,0 +1,282 @@
+    {
+        return false;
+    }
+
+    private boolean canPaneConnectToBlock(final Block blockIn)
+    {
+        return blockIn.getDefaultState().isFullCube()
+                || blockIn == this || blockIn == Blocks.GLASS || blockIn == Blocks.STAINED_GLASS ||
+                blockIn == Blocks.STAINED_GLASS_PANE || blockIn instanceof BlockPaperwall;
+    }
+
+    @SideOnly(Side.CLIENT)
+    public boolean shouldSideBeRendered(@NotNull final IBlockState blockState,
+                                        @NotNull final IBlockAccess blockAccess,
+                                        @NotNull final BlockPos pos,
+                                        @NotNull final EnumFacing side)
+    {
+        return blockAccess.getBlockState
+                (pos.offset(side)).getBlock() != this && super.shouldSideBeRendered(blockState, blockAccess, pos, side);
+    }
+
+    @NotNull
+    @SideOnly(Side.CLIENT)
+    public BlockRenderLayer getBlockLayer()
+    {
+        return BlockRenderLayer.TRANSLUCENT;
+    }
+
+    /**
+     * Convert the BlockState into the correct metadata value
+     */
+    public int getMetaFromState(@NotNull final IBlockState state)
+    {
+        return 0;
+    }
+
+    /**
+     * Returns the blockstate with the given rotation from the passed blockstate. If inapplicable, returns the passed
+     * blockstate.
+     */
+    @NotNull
+    public IBlockState withRotation(@NotNull final IBlockState Tempstate,
+                                    @NotNull final Rotation rot)
+    {
+        switch (rot)
+        {
+            case CLOCKWISE_180:
+                return Tempstate.withProperty
+                        (NORTH, Tempstate.getValue(SOUTH)).withProperty
+                        (EAST, Tempstate.getValue(WEST)).withProperty
+                        (SOUTH, Tempstate.getValue(NORTH)).withProperty
+                        (WEST, Tempstate.getValue(EAST));
+            case COUNTERCLOCKWISE_90:
+                return Tempstate.withProperty
+                        (NORTH, Tempstate.getValue(EAST)).withProperty
+                        (EAST, Tempstate.getValue(SOUTH)).withProperty
+                        (SOUTH, Tempstate.getValue(WEST)).withProperty
+                        (WEST, Tempstate.getValue(NORTH));
+            case CLOCKWISE_90:
+                return Tempstate.withProperty
+                        (NORTH, Tempstate.getValue(WEST)).withProperty
+                        (EAST, Tempstate.getValue(NORTH)).withProperty
+                        (SOUTH, Tempstate.getValue(EAST)).withProperty
+                        (WEST, Tempstate.getValue(SOUTH));
+            default:
+                return Tempstate;
+        }
+    }
+
+    /**
+     * Returns the blockstate with the given mirror of the passed blockstate. If inapplicable, returns the passed
+     * blockstate.
+     */
+    @NotNull
+    public IBlockState withMirror(@NotNull final IBlockState Tempstate, final @NotNull Mirror mirrorIn)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)